lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -431,18 +455,50 @@ def create_coefficients_cube(
             coeff_names, long_name="coefficient_name", units="no_unit")
         dim_coords_and_dims = [(coefficient_index, 0)]
         aux_coords_and_dims = [(coefficient_name, 0)]
-        for coord_name in (
-                ["time", "forecast_period", "forecast_reference_time"]):
-            try:
-                aux_coords_and_dims.append(
-                    (current_forecast.coord(coord_name), None))
-            except CoordinateNotFoundError:
-                pass
-        attributes = {"diagnostic_standard_name": current_forecast.name()}
-        for attribute in current_forecast.attributes.keys():
+
+        # Create a forecast_reference_time coordinate.
+        frt_point = cycletime_to_datetime(self.current_cycle)
+        try:
+
+            frt_coord = (
+                historic_forecast.coord("forecast_reference_time").copy(
+                    datetime_to_iris_time(frt_point, time_units="seconds")))
+        except CoordinateNotFoundError:
+            pass
+        else:
+            aux_coords_and_dims.append((frt_coord, None))
+
+        # Create forecast period and time coordinates.
+        try:
+            fp_point, = (
+                np.unique(historic_forecast.coord("forecast_period").points))
+            fp_coord = (
+                historic_forecast.coord("forecast_period").copy(fp_point))
+        except CoordinateNotFoundError:
+            pass
+        else:
+            aux_coords_and_dims.append((fp_coord, None))
+            if historic_forecast.coords("time"):
+                frt_point = cycletime_to_datetime(self.current_cycle)
+                # Ensure that the fp_point is determined with units of seconds.
+                copy_of_fp_coord = (
+                    historic_forecast.coord("forecast_period").copy())
+                copy_of_fp_coord.convert_units("seconds")
+                fp_point, = np.unique(copy_of_fp_coord.points)
+                time_point = (
+                    frt_point + datetime.timedelta(seconds=float(fp_point)))
+                time_point = datetime_to_iris_time(
+                    time_point,
+                    time_units=str(historic_forecast.coord("time").units))
+                time_coord = historic_forecast.coord("time").copy(time_point)
+                aux_coords_and_dims.append((time_coord, None))
+
+        attributes = {"diagnostic_standard_name": historic_forecast.name()}
```

Comment:
Have you got a test that checks for this attribute?

Reply:
Yes, I think as a result of doing assertEqual on the result and expected cubes.